ark/spLDA/processing_test.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import pytest
+
+import ark.spLDA.processing as pros
+import ark.settings as settings
+from ark.utils.misc_utils import verify_in_list
+from ark.utils.test_utils import make_cell_table
+
+# Generate, format, and featurize a test cell table
+N_CELLS = 1000
+TRAIN_FRAC = 0.75
+TRAIN_CELLS = TRAIN_FRAC * N_CELLS
+TEST_CELL_TABLE = make_cell_table(N_CELLS)
+TEST_FORMAT = pros.format_cell_table(
+    cell_table=TEST_CELL_TABLE, clusters=list(np.unique(TEST_CELL_TABLE[settings.CLUSTER_ID])))
+TEST_FEATURES = pros.featurize_cell_table(cell_table=TEST_FORMAT, train_frac=TRAIN_FRAC)
+
+
+def test_format_cell_table():
+    # Check that number of FOVS match
+    fov_list=[x for x in TEST_FORMAT.keys() if x not in ['fovs', 'markers', 'clusters']]
+    verify_in_list(
+        fovs1=list(np.unique(TEST_CELL_TABLE[settings.FOV_ID])), fovs2=fov_list)
+    # Check that columns were retained/renamed
+    verify_in_list(
+        cols1=["x", "y", "cluster_id", "cluster", "is_index"], cols2=list(TEST_FORMAT[1].columns))
+    # Check that columns were dropped
+    assert len(TEST_CELL_TABLE.columns) > len(TEST_FORMAT[1].columns)
+
+
+def test_featurize_cell_table():
+    # Check for consistent dimensions
+    assert TEST_FEATURES["featurized_fovs"].shape[0] == TEST_CELL_TABLE.shape[0]
```

Comment:
Since there is logic in featurize_cell_table that changes based on input arguments, we should check this. Specifically, the featurization type. We want to make sure that passing `marker`, `avg_marker`, `cluster`, etc actually works, and also that the behavior changes appropriately.

We should also add input validation to `featurize_cell_table` to make sure only valid values are provided for this argument